plugins/lookup/get_moid.py

```diff
@@ -0,0 +1,579 @@
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+lookup: get_moid
+short_description: Look up MoID for vSphere objects using vCenter REST API
+description:
+    - Returns Managed Object Reference (MoID) of the object contained in the specified path.
+options:
+    _terms:
+        type: dict
+        required: true
+        suboptions:
+            object_type:
+                description:
+                    - Type of the vSphere object to be queried
+                required: true
+                type: str
+            path:
+                description:
+                    - Type of the vSphere object to be queried
+                required: true
+                type:str
+    vcenter_hostname:
+        description:
+            - The hostname or IP address of the vSphere vCenter.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_HOST) will be used instead.
+        required: true
+        type: str
+    vcenter_password:
+        description:
+            - The vSphere vCenter password.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_PASSWORD) will be used instead.
+        required: true
+        type: str
+    vcenter_rest_log_file:
+        description:
+            - You can use this optional parameter to set the location of a log file.
+            - This file will be used to record the HTTP REST interaction.
+            - The file will be stored on the host that run the module.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_REST_LOG_FILE) will be used instead.
+        type: str
+    vcenter_username:
+        description:
+            - The vSphere vCenter username.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_USER) will be used instead.
+        required: true
+        type: str
+    vcenter_validate_certs:
+        default: true
+        description:
+            - Allows connection when SSL certificates are not valid. Set to C(false) when
+              certificates are not trusted.
+            - If the value is not specified in the task, the value of environment variable
+              C(VMWARE_VALIDATE_CERTS) will be used instead.
+        type: bool
+author:
+    - Alina Buzachis <@alinabuzachis>
+version_added: 1.0.1
+requirements:
+    - vSphere 7.0.2 or greater
+    - python >= 3.6
+    - aiohttp
+'''
+
+
+EXAMPLES = r'''
+# lookup sample
+
+- name: set connection info
+  set_fact:
+    connection_args:
+        vcenter_hostname: "vcenter.test"
+        vcenter_username: "administrator@vsphere.local"
+        vcenter_password: "1234"
+
+- name: lookup MoID of the object
+  debug: msg="{{ lookup('get_moid',
```

Comment:
I think we can simplify the syntax if we provide one plugin per resource type. This would allow us to drop the first argument. e.g:

`lookup('vmware.vmware_rest.vm_id', '/my_dc/host/my_cluster/esxi1.test/test_vm1')`
`lookup('vmware.vmware_rest.cluster_id', '/my_dc/host/my_cluster/my_cluster')`
etc

Reply:
I think having a lookup plugin for each resource type (vm, folder, resource-pool, cluster, datastore, datacenter, network) would introduce redundancy and would also require to re-think the logic of this plugin itself. With this plugin you can actually handle all these resources (vm, folder, resource-pool, cluster, datastore, datacenter, network). The only requirement is to have  `lookup('vmware.vmware_rest.get_moid', {'resource_type': 'path})`. You can also specify one or more objects to get the MoID.

Reply:
If it's not a matter of redundancy but it's related to the fact you want to autogenerate if for each resource type, I agree we can remove the `resource_type`.

Reply:
We can also move most of the logic in a module_utils file and just expose ad-hoc filter plugins. This may be actually a nice first step to prepare the ground for a potential inventory plugin later.

Reply:
Ok, thank you. I like this idea.

Reply:
Done this.